File: 342-power-of-four/342-power-of-four.java
class Solution {
    public boolean isPowerOfFour(int n) {

        for(int i=0;Math.pow(4,i)<=n;i++){
            if(n<0)
                return false;
            if(Math.pow(4,i)==n)
                return true;
            if(Math.pow(4,i)>n)
                return false;
        }
        return false;
    }
}

File: 342-power-of-four/342-power-of-four.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        int c=0,i=0,flag=1;
        while(n>0){
			   if(n&1){ c++;
							if(i%2==0)
							flag=1;
							else 
							{flag=0;
							break; }   
						}
            if(c>1) 
			break;
            n=n>>1;
			i++; }
        if(c==1&&flag==1) 
		return true;
        else 
		return false;
    }
};